exonum/src/explorer/mod.rs

```diff
@@ -0,0 +1,976 @@
+// Copyright 2017 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Blockchain explorer module provides api for getting information about blocks and transactions
+//! from the blockchain.
+
+use serde::{Serialize, Serializer};
+
+use crypto::{CryptoHash, Hash};
+use blockchain::{Schema, Blockchain, Block, TxLocation, Transaction, TransactionResult,
+                 TransactionError, TransactionErrorType};
+use encoding;
+use helpers::Height;
+use messages::{Precommit, RawMessage};
+use storage::{ListProof, Snapshot};
+
+use std::cell::{Ref, RefCell};
+use std::collections::Bound;
+use std::fmt;
+use std::ops::{Deref, Index, Range, RangeFrom, RangeFull, RangeTo};
+
+#[cfg(any(test, feature = "doctests"))]
+#[doc(hidden)]
+pub mod tests;
+
+/// Transaction parsing result.
+type ParseResult = Result<Box<Transaction>, encoding::Error>;
+
+/// Range of `Height`s.
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub struct HeightRange(pub Bound<Height>, pub Bound<Height>);
```

Comment:
Could be accomplished with [`RangeArgument`](https://doc.rust-lang.org/std/collections/range/trait.RangeArgument.html), but it's unstable as of now.